sunpy/net/helioviewer.py

```diff
@@ -450,3 +433,18 @@ def _request(self, params):
     def _format_date(self, date):
         """Formats a date for Helioviewer API requests"""
         return parse_time(date).isot + "Z"
+
+    def _get_source_id(self, source_id, key):
+        """Returns source_id based on the key"""
+        if source_id is None:
```

Comment:
Why does it check if its None?

Reply:
We need to check this because if the user passes the source_id then we don't need to calculate it using our private function

Reply:
But this function should calculate a souirce_id, not check if we need one.

Reply:
It do calculates the source_id but what if the user has passed it explicitly? Then it needs to return the same back to the function

Reply:
But if the user has passed in one, we do not need to call this function.

Reply:
Then we need a check in the function from which this private function is being called. In that case we need to add checks at two places so I thought to add it here

Reply:
I don't see why we should call out to a function only to have it return the same value.

Reply:
But we do need to add the check somewhere. We can add it in the calling function

Reply:
We can check it before we call the function.

Reply:
Ok no problem I'll add it there